app/src/main/java/com/example/sharewhatyoucanproject/donor/AddDonorFragment.kt

```diff
@@ -0,0 +1,200 @@
+
+    lateinit var addDonorViewModel: AddDonorViewModel
+    lateinit var locationRequest: LocationRequest
+    lateinit var arrayAdapter: ArrayAdapter<String>
+
+    lateinit var circularProgressIndicator: CircularProgressIndicator
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        addDonorViewModel = ViewModelProvider(
+            this,
+            AddDonorViewModelFactory(),
+        )[AddDonorViewModel::class.java]
+    }
+
+        override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?,
+        ): View? {
+            _binding = FragmentAddDonorBinding.inflate(inflater, container, false)
+
+            circularProgressIndicator = binding.progressCircular
+            locationRequest = LocationRequest.create()
+            (activity as AppCompatActivity?)!!.supportActionBar!!.hide()
+
+            addDonorViewModel.currentLocation.observe(viewLifecycleOwner) {
+                it?.let {
+                    addDonorViewModel.uploadImage()
+                }
+            }
+            addDonorViewModel.imageUrl.observe(viewLifecycleOwner) {
+                it?.let { addDonorViewModel.saveData() }
+            }
+
+            addDonorViewModel.donorResult.observe(viewLifecycleOwner) { donorResult ->
+                when (donorResult) {
+                    is DonorResult.Success -> {
+                        requireContext().showToast("Post Added")
+                        findNavController().navigateUp()
+                    }
+                    is DonorResult.Error -> {
+                        circularProgressIndicator.visibility = View.GONE
+                        requireContext().showToast(donorResult.message)
+                    }
+                }
+            }
+
+            binding.foodimg.setOnClickListener {
+                val intent = Intent()
+                intent.type = "image/*"
+                intent.action = Intent.ACTION_GET_CONTENT
+                startActivityForResult(Intent.createChooser(intent, "Select Picture"), 0)
+            }
+
+            arrayAdapter = ArrayAdapter<String>(
+                requireContext(),
+                android.R.layout.simple_spinner_dropdown_item,
+                foodTypes,
+            )
+            binding.spinner.adapter = arrayAdapter
+            binding.spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>?,
+                    view: View,
+                    position: Int,
+                    id: Long,
+                ) {
+                    foodTypes[position].also {
+                        addDonorViewModel.foodType = it
+                        requireContext().showToast(it)
+                    }
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {}
```

Comment:
should you null out the `foodType` when nothing is selected?

Reply:
done